tools/telemetry-generator/src/commands/appInsights.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "fs";
+import path from "path";
+import { Command, Flags } from "@oclif/core";
+import * as appInsight from 'applicationinsights';
+
+export class EntryPoint extends Command {
+    static flags = {
+        help: Flags.help(),
+        handlerModule: Flags.string({
+            char: "m",
+            required: true,
+            description: "Absolute path to a JavaScript file that exports a handler function to process the files " +
+                "contained in the folders specified with --dir.",
+        }),
+        dir: Flags.string({
+            char: "d",
+            multiple: true,
+            required: true,
+            description: "Folder that contain the test output files to process. " +
+                "Files in subfolders are also processed. Can be specified multiple times.",
+        }),
+        connectionString: Flags.string({
+            char: "c",
+            required: true,
+            description: "The connection string to initialize the azure telemetry client with",
+        }),
+    };
+
+    static examples = [
+        {
+            command: "$ node bin/run --handlerModule /path/to/my/module.js --dir /path/to/my/files",
+            description: "Process files from /path/to/my/files and all its subfolders, using the handler at " +
+                "/path/to/my/module.js",
+        },
+    ];
+
+    async run() {
+        const { flags } = await this.parse(EntryPoint);
+
+        let handler;
+        try {
+            // Note: we expect the path to the handler module to be absolute. Relative paths technically work, but
+            // one needs to be very familiar with Node's module resolution strategy and understand exactly which file
+            // is the one getting executed at runtime (since that's where the relative path will be resolved from).
+            handler = (await import(flags.handlerModule)).default;
+        } catch (err) {
+            exitWithError(`Unexpected error importing specified handler module.\n${err}`);
+        }
+
+        if (typeof handler !== "function") {
+            exitWithError("Handler module does not have a function as its default export");
+        }
+
+        console.log(`obtained connectionstring: ${flags.connectionString}`)
```

Comment:
I'd say probably remove? Locally we're passing it to the command, and in ADO it gets replaced with *** for security.

Reply:
Yeah this was for debugging. Removed in recent commit # 84045